jib-plugins-common/src/main/java/com/google/cloud/tools/jib/plugins/common/AnsiLoggerWithFooter.java

```diff
@@ -79,12 +86,26 @@ public void awaitTermination() {
   /**
    * Runs {@code messageLogger} asynchronously.
    *
-   * @param messageLogger the {@link Runnable} intended to synchronously log a message to the
-   *     console
+   * @param messageLogger the {@link Consumer} intended to synchronously log a message to the
+   *     console. {@code messageLogger} should print a new line at the end.
+   * @param message the message to log with {@code messageLogger}
    * @return a {@link Future} to track completion
    */
-  public Future<Void> log(Runnable messageLogger) {
-    return log(messageLogger, footerLines);
+  public Future<Void> log(Consumer<String> messageLogger, String message) {
+    return executorService.submit(
+        () -> {
+          boolean didErase = eraseFooter();
+
+          // If a previous footer was erased, the message needs to go up a line.
+          String messagePrefix = didErase ? CURSOR_UP_SEQUENCE : "";
+          messageLogger.accept(messagePrefix + message);
```

Comment:
Will this work when `messageLogger` itself is prepending, e.g., "[INFO] "? For example, I wonder if the end string might be "[INFO] + <cursor up> + message", in which case you leave "[INFO]" at the bottom and the actual message line loses "[INFO]".

Reply:
Yep, it works since cursor up goes up in the same column. However, it does replace other message lines' prepended text like `[DEBUG]` with `[INFO]`.

Reply:
I see. I think we can tolerate that.